swervedrive/icr/kinematicmodel.py

```diff
@@ -132,3 +129,56 @@ def reconfigure_wheels(self, beta_d: np.ndarray, beta_e: np.ndarray):
         if np.isclose(dbeta, 0, atol=1e-2).all():
             self.state = KinematicModel.State.RUNNING
         return dbeta
+
+    def compute_odometry(self, lmda_e: np.ndarray, mu_e: float, delta_t: float):
+        """
+        Update our estimate of epsilon (twist position) based on the new ICR
+        estimate.
+        :param lmda_e: the estimate of the ICR in h-space.
+        :param mu_e: estimate of the position of the robot about the ICR.
+        :param delta_t: time since the odometry was last updated.
+        """
+        xi_dot = mu_e * np.array([[lmda_e[1]], [-lmda_e[0]], [lmda_e[2]]])  # Eq (2)
+        theta = self.xi[0, 2]
+        m3 = np.array(
+            [
+                [np.cos(theta), -np.sin(theta), 0],
+                [np.sin(theta), np.cos(theta), 0],
+                [0, 0, 1],
+            ]
+        )
+        self.xi += np.matmul(m3, xi_dot).T * delta_t  # Eq (24)
+        return self.xi
+
+    def estimate_mu(self, phi_dot: np.ndarray, lmda_e):
+        """
+        Find the rotational position of the robot about the ICR.
+        :param phi_dot: array of angular velocities of the wheels.
+        :param lmda_e: the estimate of the ICR in h-space.
+        :returns: the estimate of mu (float).
+        """
+        # this requires solving equation (22) from the control paper, i think
+        # we may need to look into whether this is valid for a system with no
+        # wheel coupling
+        lmda_e = np.reshape(lmda_e, (len(lmda_e), 1))
+        phi_dot = np.reshape(phi_dot, (len(phi_dot), 1))
```

Comment:
This is not a very safe way of reshaping. Can lead to issues like:
```
(ins)>>> import numpy as np
(ins)>>> a = np.array([[3, 4, 5]])
(ins)>>> a
array([[3, 4, 5]])
(ins)>>> len(a)
1
```
We should be using (-1, 1) instead.